File: app/src/main/java/com/test/test168/juhe/HealthNewsDetailsActivity.java
package com.test.test168.juhe;

import android.support.v7.widget.Toolbar;
import android.text.Html;
import android.widget.TextView;

import com.test.test168.R;
import com.test.test168.TConstants;
import com.test.test168.base.BaseActivity;
import com.test.test168.bean.JuheHealthNewsClassListItem;
import com.test.test168.bean.JuheHealthNewsDetails;
import com.xian.common.utils.XLog;

import butterknife.BindView;
import butterknife.ButterKnife;

public class HealthNewsDetailsActivity extends BaseActivity {

    JuheHealthNewsClassListItem newsClass;
    HealthNewsDetailsLoader mLoader;

    @BindView(R.id.detail_toolbar)
    Toolbar detailToolbar;
    @BindView(R.id.tv_news_details_time)
    TextView tvNewsDetailsTime;
    @BindView(R.id.tv_news_details_keyword)
    TextView tvNewsDetailsKeyword;
    @BindView(R.id.tv_news_details_read_count)
    TextView tvNewsDetailsReadCount;
    @BindView(R.id.tv_news_details_content)
    TextView tvNewsDetailsContent;

    @Override
    protected void initViews() {
        setContentView(R.layout.activity_health_news_details);
        ButterKnife.bind(this);
        newsClass = (JuheHealthNewsClassListItem) getIntent().getSerializableExtra(TConstants.IntentKey.HEALTH_DETAILS);
        if (newsClass != null)
            setViews(newsClass);
    }

    private void setViews(JuheHealthNewsClassListItem newsClass) {
        showLoadingDialog();
        detailToolbar.setTitle(newsClass.getTitle());
        mLoader = new HealthNewsDetailsLoader(mActivity);
        mLoader.loader(newsClass,
                new CustomJuheSub<JuheHealthNewsDetails>() {
                    @Override
                    protected void onSuccess(JuheHealthNewsDetails result) {
                        XLog.i("on success : " + result);
                        setViewDetails(result);
                        dismissLoadingDialog();
                    }

                    @Override
                    protected void onFailure(String errorMsg) {
                        XLog.e(errorMsg);
                        dismissLoadingDialog();
                    }
                });
    }

    private void setViewDetails(JuheHealthNewsDetails result) {
        tvNewsDetailsTime.setText("时间：" + new java.sql.Date(result.getTime()).toString());
        tvNewsDetailsKeyword.setText("关键字：" + result.getKeywords());
        tvNewsDetailsReadCount.setText("阅读次数：" + result.getCount());
        tvNewsDetailsContent.setText(Html.fromHtml(result.getMessage()));
    }
}

File: app/src/main/java/com/test/test168/juhe/HealthNewsDetailsActivity.kt
package com.test.test168.juhe

import androidx.appcompat.widget.Toolbar
import android.text.Html
import android.widget.TextView

import com.test.test168.R
import com.test.test168.TConstants
import com.test.test168.base.BaseActivity
import com.test.test168.bean.JuheHealthNewsClassListItem
import com.test.test168.bean.JuheHealthNewsDetails
import com.xian.common.utils.XLog


class HealthNewsDetailsActivity : BaseActivity() {

    internal var newsClass: JuheHealthNewsClassListItem? = null
    internal var mLoader: HealthNewsDetailsLoader? = null

    internal var detailToolbar: Toolbar? = null
    internal var tvNewsDetailsTime: TextView? = null
    internal var tvNewsDetailsKeyword: TextView? = null
    internal var tvNewsDetailsReadCount: TextView? = null
    internal var tvNewsDetailsContent: TextView? = null

    override fun initViews() {
        setContentView(R.layout.activity_health_news_details)

        detailToolbar = findViewById(R.id.detail_toolbar)
        tvNewsDetailsTime = findViewById(R.id.tv_news_details_time)
        tvNewsDetailsKeyword = findViewById(R.id.tv_news_details_keyword)
        tvNewsDetailsReadCount = findViewById(R.id.tv_news_details_read_count)
        tvNewsDetailsContent = findViewById(R.id.tv_news_details_content)

        newsClass = intent.getSerializableExtra(TConstants.IntentKey.HEALTH_DETAILS) as JuheHealthNewsClassListItem
        if (newsClass != null)
            setViews(newsClass!!)
    }

    private fun setViews(newsClass: JuheHealthNewsClassListItem) {
        showLoadingDialog()
        detailToolbar!!.title = newsClass.title
        mLoader = HealthNewsDetailsLoader(mActivity)
        mLoader?.loader(newsClass,
                object : CustomJuheSub<JuheHealthNewsDetails>() {
                    override fun onSuccess(result: JuheHealthNewsDetails) {
                        XLog.i("on success : " + result)
                        setViewDetails(result)
                        dismissLoadingDialog()
                    }

                    override fun onFailure(errorMsg: String) {
                        XLog.e(errorMsg)
                        dismissLoadingDialog()
                    }
                })
    }

    private fun setViewDetails(result: JuheHealthNewsDetails) {
        tvNewsDetailsTime!!.text = "时间：" + java.sql.Date(result.time).toString()
        tvNewsDetailsKeyword!!.text = "关键字：" + result.keywords
        tvNewsDetailsReadCount!!.text = "阅读次数：" + result.count
        tvNewsDetailsContent!!.text = Html.fromHtml(result.message)
    }
}
